101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "projectName": {
+            "type": "string",
+            "defaultValue": "GEN-UNIQUE",
+            "metadata": {
+                "description": "Define the project name or prefix for all objects."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]"
+        },
+        "DNSEntry": {
+            "type": "string",
+            "defaultValue": "remoteapps",
+            "metadata": {
+                "description": "The name for web access and gateway."
+            }
+        },
+        "DNSZone": {
+            "type": "string",
+            "defaultValue": "contosocorp.com",
+            "metadata": {
+                "description": "External public DNS domain zone. NOT AD domain."
+            }
+        },
+        "DNSZoneResourceGroup": {
+            "type": "string",
+            "defaultValue": "rgdns",
+            "metadata": {
+                "description": "Resource group name for external dns zone to modify, if 'CreateDNSEntry' is true."
+            }
+        },
+        "createDNSEntry": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "Do you want to create/update the DNS entries on Azure DNS domain?"
+            }
+        },
+        "ExternalFqdn": {
+            "type": "string",
+            "defaultValue": "[concat(parameters('DNSEntry'),'.',parameters('DNSZone'))]",
+            "metadata": {
+                "description": "Fully qualified external domain name. eg. remoteapps.contoso.com"
+            }
+        },
+        "TimeZoneID": {
+            "type": "string",
+            "defaultValue": "UTC",
+            "metadata": {
+                "description": "TimeZone ID to be used on VMs. Get available timezones with powershell Get-TimeZone command."
+            }
+        },
+       "vmNames": {

```

Comment:
prompting for names here will create a scenario where things that need to be globally unique are not (e.g. dns) - and manually entering arrays as params is not for humans.  Any reason this cannot be a variable?  Then you can guarantee uniqueness.